### ISSUE TITLE
Enabling TLS in iframe

### ISSUE THREAD

**Issue by fanfavorite** [silkimen/cordova-plugin-advanced-http#52]
This plugin works great to enable TLS for post and get functions for Android 4.1+. My issue now is that I need to enable TLS for iframes as well. Can I use this plugin to somehow do that?

**Comment by silkimen** [comment]
Hi fanfavorite,
you'll need to implement a "bridge" between your main window and the iframe. You can achieve this by using [cross-document messaging](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage).

**Comment by fanfavorite** [comment]
Thanks silkimen. I have used postMessage before, but not sure how you would use that to enable TLS 1.2? Any help would be appreciated.

**Comment by silkimen** [comment]
You can use this plugin only in your main window (the cordova-enabled one), but you need to interact with it out of your iframe. So the idea is to create some kind of "bridge" between these two.

e.g. in your iframe you would do something like:
```js
myBridge.http.get('http://example.com/', {}, {}, onSuccess, onError);
```

which will call the plugin on the main window like:
```js
cordova.plugin.http.get('https://google.com/', {}, {}, onSuccess, onError);
```

So `myBridge` needs to implement functionality, which is accepting parameters for the cordova plugin and passing to the main window.

So here's some ES6 example code, just to get an idea. Neither tested nor complete and without any exception handling.

In iframe:
```js
const myBridge = {
  currentSequence: 0,
  handlers: {},

  init: () => window.addEventListener('message', myBridge.onMessage, false),

  onMessage: event => {
    const handler = myBridge.handlers[event.data.sequenceId];

    if (event.data.type === 'success') {
      handler.success(event.data.payload);
    } else {
      handler.error(event.data.payload);
    }

    delete myBridge.handlers;
  },

  dispatchFnCall: (data, success, error) => {
     myBridge.currentSequence += 1;
     myBridge.handlers[myBridge.currentSequence] = { success, error };
     data.sequenceId = myBridge.currentSequence;
     mainWindow.postMessage(data);
  }
};

myBridge.http = {
  get: (url, params, headers, success, error) =>
    myBridge.dispatchFnCall({ method: 'get', url, params, headers }, success, error),

  ......
};
```

In main window:

```js
window.addEventListener('message', event => {
  const { sequenceId, method, url, params, headers } = event.data;

  const onSuccess = response =>
    event.source.postMessage({ sequenceId, type: 'success', payload: response }, event.origin);

  const onError = error =>
    event.source.postMessage({ sequenceId, type: 'error', payload: error }, event.origin);

  cordova.plugin.http[method](url, params, headers, onSuccess, onError);
}, false);
```

So basically the idea is to track communication between main window and iframe by an sequence ID which will be returned on any message received from the main window. We need this ID to track which handler we need to run in the iframe. I hope you get the idea.

**Comment by fanfavorite** [comment]
Thanks. Appreciate the code examples and explanation. I understand this way to call different functions, however when the iframe URL is blocked due to TLS 1.2, that won't trigger the myBridge code as its already denied trying to access the page. Am I missing something for that? I realize I wasn't as clear as I should have been above, so I apologize for that. I was thinking I could populate the iframe using the get method in the main window, however every URL in that returned code (scripts, stylesheets, images, etc) will be blocked, which isn't very manageable.

**Comment by silkimen** [comment]
Ah ok, now I understand your problem!

Unfortunately, you won't be able to use this plugin to achieve a solution for this scenario. You'd need to change the default request configuration of the cordova webview. I don't know if this is possible and how to do. Sorry!

**Comment by silkimen** [comment]
Closing this, hope you'll find another solution for your problem!
